src/main/groovy/org/pih/warehouse/inventory/CycleCountRequestBatchCommand.groovy

```diff
@@ -0,0 +1,28 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.Validateable
+import org.pih.warehouse.product.Product
+
+class CycleCountRequestBatchCommand implements Validateable {
+
+    List<CycleCountRequestCommand> requests
+
+    static constraints = {
+        requests(validator: { List<CycleCountRequestCommand> requests ->
+            // First catch if any duplicates have been provided (e.g. 10006, 10006 in the payload)
+            // This doesn't check though if any duplicate is already persisted, this is checked in the CCR command below (.each + .validate() clause)
+            Map<String, List<Product>> duplicates = requests.product
+                    .groupBy { it.productCode }
+                    .findAll { it.value.size() > 1 }
```

Comment:
you could probably do a countBy productCode (and then filter to > 1) so that you only need to maintain a List<String> (of duplicate productCodes) instead of a map but that's a tiny optimization so I'm not worried about it.